tests/rptest/tests/redpanda_oauth_test.py

```diff
@@ -7,47 +7,58 @@
 # the Business Source License, use of this software will be governed
 # by the Apache License, Version 2.0
 
-import time
-import functools
-import json
-
+from ducktape.tests.test import Test
+from rptest.clients.rpk import RpkTool
 from rptest.clients.python_librdkafka import PythonLibrdkafka
-from rptest.services.redpanda import SecurityConfig, make_redpanda_service
+from rptest.services.redpanda import LoggingConfig, SecurityConfig, make_redpanda_service
 from rptest.services.keycloak import KeycloakService
 from rptest.services.cluster import cluster
-
-from ducktape.tests.test import Test
-from rptest.services.redpanda import make_redpanda_service
-from rptest.clients.rpk import RpkTool
 from rptest.util import expect_exception
-
 from confluent_kafka import KafkaException
 
 CLIENT_ID = 'myapp'
+TOKEN_AUDIENCE = 'account'
 EXAMPLE_TOPIC = 'foo'
 
+log_config = LoggingConfig('info',
+                           logger_levels={
+                               'security': 'trace',
+                               'pandaproxy': 'trace',
+                               'kafka/client': 'trace',
+                               'http': 'trace'
+                           })
+
 
 class RedpandaOIDCTestBase(Test):
     """
     Base class for tests that use the Redpanda service with OIDC
     """
     def __init__(self,
                  test_context,
-                 num_nodes=5,
+                 num_nodes=4,
                  sasl_mechanisms=['SCRAM', 'OAUTHBEARER'],
                  **kwargs):
         super(RedpandaOIDCTestBase, self).__init__(test_context, **kwargs)
         self.produce_messages = []
         self.produce_errors = []
         num_brokers = num_nodes - 1
         self.keycloak = KeycloakService(test_context)
+        kc_node = self.keycloak.nodes[0]
+        try:
+            self.keycloak.start_node(kc_node)
+        except Exception as e:
+            self.logger.error(f"{e}")
+            self.keycloak.clean_node(kc_node)
+            assert False, "Keycloak failed to start"
 
         security = SecurityConfig()
         security.enable_sasl = True
         security.sasl_mechanisms = sasl_mechanisms
 
-        self.redpanda = make_redpanda_service(test_context, num_brokers)
-        self.redpanda.set_security_settings(security)
+        self.redpanda = make_redpanda_service(test_context,
+                                              num_brokers,
+                                              security=security,
+                                              log_config=log_config)
 
         self.su_username, self.su_password, self.su_algorithm = self.redpanda.SUPERUSER_CREDENTIALS
```

Comment:
I distinctly recall making this choice consciously, but I can't find anything in my notes 😞 

It's worth considering that this would have been my first exposure to ducktape (so the reason might have been bogus) and that the test has changed shape somewhat (so if valid at the time it might be irrelevant).

Reply:
> Question: Any reason why this isn't a child class of `RedpandaTest`?

Deferred construction of RedpandaService so that the `discovery_url` is available at startup of Redpanda.